src/main/java/com/internet/mixshop/model/User.java

```diff
@@ -30,6 +31,30 @@ public User(Long id, String name, String login, String password, Set<Role> roles
         this.roles = roles;
     }
 
+    public User(Long id, String name, String login, String password, byte[] salt) {
+        this.id = id;
+        this.name = name;
+        this.login = login;
+        this.password = password;
+        this.salt = salt;
+    }
+
+    public User(String name, String login, String hashPassword, byte[] salt) {
+        this.name = name;
+        this.login = login;
+        this.password = hashPassword;
+        this.salt = salt;
+    }
+
+    public User(Long id, String name, String login, String password, byte[] salt, Set<Role> roles) {
+        this.id = id;
+        this.name = name;
+        this.login = login;
+        this.password = password;
+        this.salt = salt;
+        this.roles = roles;
+    }
```

Comment:
![image](https://user-images.githubusercontent.com/50640794/94825379-c7055200-040e-11eb-845b-3d8be3ed7393.png)
modify your constructors like this